/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

import "noisicaa/audioproc/public/musical_time.proto";
import "noisicaa/core/proto_types.proto";

package noisicaa.pb;

message SessionValue {
  required string name = 1;

  oneof type {
    string string_value = 2;
    bytes bytes_value = 3;
    bool bool_value = 4;
    int64 int_value = 5;
    double double_value = 6;
    Fraction fraction_value = 7;
    MusicalTime musical_time_value = 8;
    MusicalDuration musical_duration_value = 9;
  }
}

message SessionDataCheckpoint {
  repeated SessionValue session_values = 1;
}
